### PR TITLE
Refactor: SDR & SI_SDR

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Unify the notation and keep more descriptive functional names
part of #729

## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -56,6 +56,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Renamed IoU -> Jaccard Index ([#662](https://github.com/PyTorchLightning/metrics/pull/662))
 
 
+- Renamed audio SDR metrics: ([#711](https://github.com/PyTorchLightning/metrics/pull/711))
+  * `functional.sdr` -> `functional.signal_distortion_ratio`
+  * `functional.si_sdr` -> `functional.scale_invariant_signal_distortion_ratio`
+  * `SI_SDR` -> `ScaleInvariantSDR`
```
Comment (justusschock) [inline CHANGELOG.md:62]: Also inconsistency between class and functional interface (similar to #712 ): either sdr or signal_distortion_ratio

**Comment by quancs** [comment]
@Borda why we need to rename SDR & SISDR? SDR and SISDR are both commonly used name in audio field

**Comment by Borda** [comment]
> SDR and SISDR are both commonly used name in audio field. It's short and easy to type

that is the point that they are common in audio, but then we import them in all metrics level... :rabbit: 
also have `_` in the class name is not pythonic

**Comment by quancs** [comment]
> that is the point that they are common in audio, but then we import them in all metrics levels... 🐰 in fact, I am fine with SDR, but the SI is a bit confusing my main concern is having `_` in the class name is not pythonic and inconsistent with other metrics

Importing sdr and si_sdr in all metrics levels only make troubles when some other metrics have the same short name with them. After googling SDR, I can only find `Special Drawing Rights (SDRs) `, `software defined Radio (SDR)`, `Sales Development Representatives (SDRs)` in the first three pages. If you think it's a confusing name, can you give some examples?

Below is the original definition of SI-SDR in paper. 
图片
The `-` cannot be used, thus `_` is used instead. I know it's not pythonic, but add a separator is the tradition in audio field. I can list many and many papers to demenstrate that.

**Comment by Borda** [comment]
@quancs first of all, it is not meant as any form of offense to audio metrics or lowering your contribution! I just try to find the best way for TorchMterics as a coherent package and have a consistent name convention...

**Comment by quancs** [comment]
@Borda I was a little bit hurry before. ^^
Removing the `_` is OK for me. But I really don't like the idea to expand the names of audio metrics. The short names are unique, clear and concise in audio field.

**Comment by Borda** [comment]
@PyTorchLightning/core-metrics updated and keeping abbreviations, pls check it now :rabbit:

**Comment by Borda** [comment]
@quancs is there a reason why you are moving it back to separate files?
I thought that we (I suggested and you did not dispute) agreed to have it in the same file

**Comment by quancs** [comment]
> @quancs is there a reason why you are moving it back to separate files? I thought that we (I suggested and you did not dispute [#712 (comment)](https://github.com/PyTorchLightning/metrics/pull/712#issuecomment-1006895923)) agreed to have it in the same file

I remembered that you said you would go with 2 ( 2 put sisnr and sisdr together;), so ...

**Comment by quancs** [comment]
@Borda I thought that's our agreement ... So, you changed your mind? ^^

**Comment by quancs** [comment]
I'm OK with both choices. You can choose the anyone you like.

**Comment by quancs** [comment]
@Borda Another decision we need to make is about the full name of SDR and SNR. `signal_to_noise_ratio` or `signal_noise_ratio`? `signal_to_distortion_ratio` or `signal_distortion_ratio`? What do you think

**Comment by Borda** [comment]
> @Borda Another decision we need to make is about the full name of SDR and SNR. `signal_to_noise_ratio` or `signal_noise_ratio`? `signal_to_distortion_ratio` or `signal_distortion_ratio`? What do you think

I would follow `sklearn` and `skimage` and drite it without `_to_`

> I remembered that you said you would go with 2 ( 2 put sisnr and sisdr together;), so ...

I made typo, then fixed it, but in all conversations, I was saying to put sdr and si-sdr together as it rather a variant of sdr the si

> I'm OK with both choices. You can choose the anyone you like.

lets stick with the resolution on metrics slack

**Comment by quancs** [comment]
> > @Borda Another decision we need to make is about the full name of SDR and SNR. `signal_to_noise_ratio` or `signal_noise_ratio`? `signal_to_distortion_ratio` or `signal_distortion_ratio`? What do you think
> 
> I would follow `sklearn` and `skimage` and drite it without `_to_`
> 
> > I remembered that you said you would go with 2 ( 2 put sisnr and sisdr together;), so ...
> 
> I made typo, then fixed it, but in all conversations, I was saying to put sdr and si-sdr together as it rather a variant of sdr the si
> 
> > I'm OK with both choices. You can choose the anyone you like.
> 
> lets stick with the resolution on metrics slack

OK. Agree.

**Review by quancs — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**